components/mindstorms-ev3/src/main/java/com/example/ev3/RobotPort.java

```diff
@@ -0,0 +1,75 @@
+package com.example.ev3;
+
+import com.streambase.sb.Schema;
+
+/**
+ * Object representing a physical port on the EV3 brick. Stores information about its address and sensor type for the
+ * EV3StatusAdapter.
+ * 
+ * Copyright © 2021. TIBCO Software Inc. This file is subject to the license terms contained in the license file that is
+ * distributed with this file.
+ */
+
+public class RobotPort {
+
+    private String name;
+    private boolean streaming;
+    private SensorTypeEnum sensor;
+    private String mode;
+    private byte address;
+    private Schema schema;
+
+    public RobotPort(String name, String mode, boolean streaming, byte address, Schema schema, SensorTypeEnum sensor) {

```

Comment:
Try to keep the overloaded constructor's parameters in the same order. For example: `name`, `streaming`, `address`, and `schema` are part of each constructor so they should be the first 4 arguments. Anything extra should go after (in the same order if they are repeated, like `mode`.